robot_smach_states/src/robot_smach_states/navigation/follow_operator2_0.py

```diff
@@ -0,0 +1,475 @@
+        buffer_msg.color.g = 0
+        buffer_msg.color.b = 0
+        # buffer_msg.lifetime = rospy.Time(.0)
+        buffer_msg.id = 0
+        buffer_msg.action = Marker.ADD
+
+        for crumb in buffer:
+            buffer_msg.points.append(kdl_conversions.kdlVectorToPointMsg(crumb.pose.frame.p))
+
+        line_strip = Marker()
+        line_strip.type = Marker.LINE_STRIP
+        line_strip.scale.x = 0.05
+        line_strip.header.frame_id = "/map"
+        line_strip.header.stamp = rospy.Time.now()
+        line_strip.color.a = 1
+        line_strip.color.r = 0
+        line_strip.color.g = 1
+        line_strip.color.b = 1
+        line_strip.id = 0
+        line_strip.action = Marker.ADD
+
+        # Push back all pnts
+        for pose_stamped in ros_plan:
+            line_strip.points.append(pose_stamped.pose.position)
+
+        self._plan_marker_pub.publish(line_strip)
+        self._breadcrumb_pub.publish(buffer_msg)
+        self._robot.base.local_planner.setPlan(ros_plan, p, o)
+        userdata.buffer_follow_out = buffer
+        return 'follow_bread'
+
+
+class AskFinalize(smach.State):
+    def __init__(self, robot):
+        smach.State.__init__(self, outcomes=['follow', 'Done'])
+        self._robot = robot
+
+    def execute(self, userdata=None):
+        sentence = "Are we there yet?"
+        self._robot.speech.speak(sentence, block=True)
+        try:
+            answer = self._robot.hmi.query(sentence, "T -> yes | no", "T")
+        except TimeoutException as e:
+            self._robot.speech.speak("I did not hear you!")
+            rospy.sleep(2)
+            self._robot.speech.speak("Since I did not hear from you I assume we are done!")
+        else:
+            if answer.sentence == "yes":
+                self._robot.speech.speak("We reached our final destination!")
+                return 'Done'
+            else:
+                return 'follow'
+
+
+class Recovery(smach.State):
+    def __init__(self, robot, lost_timeout=60, lost_distance=0.8):
+        smach.State.__init__(self, outcomes=['Failed', 'follow'])
+        self._robot = robot
+        self._operator_name = "operator"
+        self._lost_timeout = lost_timeout
+        self._lost_distance = lost_distance
+        self._face_pos_pub = rospy.Publisher('/%s/follow_operator/operator_detected_face' % robot.robot_name,
+                                             geometry_msgs.msg.PointStamped, queue_size=10)
+
+    def execute(self, userdata=None):
+        rospy.loginfo("Trying to recover the operator")
+        self._robot.head.look_at_standing_person()
+        self._robot.speech.speak("%s, please look at me while I am looking for you" % self._operator_name, block=False)
+
+        # Wait for the operator and find his/her face
+        operator_recovery_timeout = self._lost_timeout
+        start_time = rospy.Time.now()
+
+        look_distance = 2.0  # magic number 4
+        look_angles = [0.0,  # magic numbers
```

Comment:
Why do you need these magic numbers? Can't you do it any other way? E.g.,
`look_angles = [-4/i * math.pi for i in range(-4, 5)]`